utils/usbmuxd/files/usbmuxd.hotplug

```diff
@@ -0,0 +1,24 @@
+case "$ACTION" in
+	bind)
+		dev=/sys$DEVPATH
+
+		[ ! -f /tmp/iPhone.lock ] && [ -d ${dev}/net ] &&
+		{
+			readlink ${dev}/driver | grep ipheth &&
+			{
+				sleep 5
+				carrier_path=${dev}/net/*/carrier
+				carrier=`cat ${carrier_path}`
+
+				[ "${carrier}" == "0" ] &&
```

Comment:
== is a bashism.

Reply:
run through shellcheck.

Reply:
Both == and = work equally well. I prefer ==. I've been using this for a very long time.

``` bash
[ "a" == "a" ] && echo yes || echo no
yes

[ "a" == "b" ] && echo yes || echo no
no

[ "a" = "a" ] && echo yes
yes

echo $SHELL
/bin/ash
```

Reply:
right, busybox has no issue with == but dash does. It's not wrong to use == but generally bashisms are frowned upon. The only exception is string substitution.

Reply:
I find it more confusing when = means both assignment (previous line) and comparison. == is well supported in the environment and better for readability. Works in `ash bash csh sh`. Indeed `dash zsh` don't like ==, and yet all shells `ash bash csh dash sh zsh` support =, so ok, I changed it ☹️

Reply:
That...is surprising that zsh doesn't like it. IIRC there are some that use zsh with OpenWrt.

Reply:
To clear the confusion I ran all the tests on macOS, intentionally with the built-in tools, because they are ancient. Usually from 2004 and if something is unsupported, it will show. Surprisingly `zsh-5.9 (x86_64-apple-darwin23.0)` is from 2022. I would assume they can't update them, due to GPL licensing? Only `ash` is from OpenWRT.